src/EVM.hs

```diff
@@ -590,7 +590,7 @@ exec1 = do
           copyBytesToMemory (the state calldata) (Lit calldatasize) (Lit 0) (Lit 0)
           executePrecompile self (the state gas) 0 calldatasize 0 0 []
           vmx <- get
-          case view (state.stack) vmx of
```

Comment:
Can this be `vm.state.stack`?

Reply:
It can be `vmx._state._stack` at the moment. I changed this line because with `OverloadedRecordDot` the function composition dot has to have spaces around it. I made as few changes as possible to enable those extensions in this PR, but I can change the rest of the trivial `view`s in the next PR.